src/service.js

```diff
@@ -292,49 +255,34 @@ export function getServicePromiseOrNull(win, id) {
   return getServicePromiseOrNullInternal(win, id);
 }
 
-
 /**
  * Returns a service for the given id and ampdoc (a per-ampdoc singleton).
  * If the service is not yet available the factory function is invoked and
  * expected to return the service.
  * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrDoc
  * @param {string} id of the service.
  * @param {function(!./service/ampdoc-impl.AmpDoc):T=} opt_factory
- *     Should create the service if it does not exist yet. If the factory is
- *     not given, it is an error if the service does not exist yet.
+ *     Factory to create the service if the service does not exist yet.
```

Comment:
Yeah, I think `performanceForOrNull` is used as you describe and thus it's needed. I mostly reacted to absence of `getExistingServiceForDoc` because I thought we were trying to remove API confusion between service registration and service use. E.g. Viewer is registered only once, but is being used via `viewerFor` all over the place. 

But, I'm ok if we reconsider this in a separate PR.